### PR TITLE
[web][SIMS #283]Student Welcome Page

### PR DESCRIPTION
**Pull request by henryoforeh-aot**
This is the new welcome dashboard page for students designed according to design specs from figma. The following navigations are expected to happen from the page.

1. When you click on "Start Application" button, you should be redirected to the page where you can begin a new application.
2. When you click on "Manage loan" button, you should be redirected to the external url of National Student Loans Service Centre (NSLSC).

student-welcome1
student-welcome2

### CODE REVIEW

File: sources/packages/web/src/views/student/StudentDashboard.vue [hunk 1]
```diff
@@ -16,22 +19,31 @@ import { useStore } from "vuex";
...
 export default {
+  emits: ["goToStudentApplication"],
   components: {
     formio,
-    FullPageContainer,
     RestrictionBanner,
     CheckValidSINBanner,
   },
   setup() {
+    const toast = useToastMessage();
```
Comment (ann-aot) [inline sources/packages/web/src/views/student/StudentDashboard.vue:34]: looks like `toast` is not used here

File: sources/packages/web/src/assets/css/formio-shared.scss [hunk 2]
```diff
@@ -376,3 +376,11 @@
   min-height: 150px;
 }
 
+.img-student-welcome-background {
+  background-image: url("../images/image_student_welcome.svg");
+  background-repeat: no-repeat;
+  background-size: contain;
+  width: 100%;
+  height: 100%;
+  min-height: 250px;
```
Comment (ann-aot) [inline sources/packages/web/src/assets/css/formio-shared.scss:385]: i see min height `250px`  used in other classes too, 
image
. maybe we can create a scss constant for it

File: sources/packages/forms/src/form-definitions/studentwelcomepage.json [hunk 3]
```diff
@@ -0,0 +1,1813 @@
...
+                                        },
+                                        {
+                                            "label": "Manage loan",
+                                            "action": "custom",
+                                            "showValidations": false,
+                                            "leftIcon": "fa fa-external-link",
+                                            "tableView": false,
+                                            "key": "manageLoan",
+                                            "type": "button",
+                                            "custom": "window.location.href = \"https://protege-secure.csnpe-nslsc.canada.ca/en/public/sign-in-method\";",
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/studentwelcomepage.json:1175]: I believe that we have been opening external links in a new tab. I missed this in previous PR for the other welcome page.

File: sources/packages/forms/src/form-definitions/studentwelcomepage.json [hunk 4]
```diff
@@ -0,0 +1,1813 @@
...
+                },
+                {
+                    "label": "Columns",
+                    "labelWidth": "",
+                    "labelMargin": "",
+                    "columns": [
+                        {
+                            "components": [
+                                {
+                                    "html": "<p>For further information, please refer to the StudentAid BC <a href=\"https://studentaidbc.ca/help-centre\"><strong>Help Centre.</strong></a></p>",
```
Comment (andrewsignori-aot) [inline sources/packages/forms/src/form-definitions/studentwelcomepage.json:1572]: I believe that we have been opening external links in a new tab. I missed this in previous PR for the other welcome page.

**Review by ann-aot — COMMENTED** [review]
LGTM 👍 Just to minor comments

File: sources/packages/web/src/views/student/StudentDashboard.vue [hunk 5]
```diff
@@ -16,22 +19,31 @@ import { useStore } from "vuex";
 import { StudentService } from "@/services/StudentService";
 import formio from "@/components/generic/formio.vue";
 import RestrictionBanner from "@/views/student/RestrictionBanner.vue";
-import FullPageContainer from "@/components/layouts/FullPageContainer.vue";
 import CheckValidSINBanner from "@/views/student/CheckValidSINBanner.vue";
+import { useRouter } from "vue-router";
+import { useToastMessage } from "@/composables";
 
 export default {
+  emits: ["goToStudentApplication"],
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/student/StudentDashboard.vue:27]: Emits is used when a component is emitting a event to a consumer. Please remove it.

File: sources/packages/web/src/views/student/StudentDashboard.vue (hunk 1 above)
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/student/StudentDashboard.vue:34]: Is this used anywhere? If not, please remove it.

File: sources/packages/web/src/views/student/StudentDashboard.vue [hunk 6]
```diff
@@ -1,12 +1,15 @@
...
     <RestrictionBanner
       v-if="hasRestriction"
       :restrictionMessage="restrictionMessage"
     />
     <CheckValidSINBanner />
-    <formio formName="studentdashboard"></formio>
-  </full-page-container>
+    <formio
+      formName="studentwelcomepage"
+      @submitted="goToStudentApplication"
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/student/StudentDashboard.vue:10]: Please avoid using the submit for this case. We have examples of other buttons executing a redirect.
Please take a look on below file to check how _customEvent_ were used.
https://github.com/bcgov/SIMS/blob/8881a35e353ebfb82f01dda7012e33ebaf86cbaa/sources/packages/web/src/components/students/ApplicationDetails.vue

**Review by andrewsignori-aot — COMMENTED** [review]
The page is looking good, please take a look at the comments 😉

**Review by guru-aot — APPROVED** [review]
LGTM

**Review by guru-aot — CHANGES_REQUESTED** [review]
Please check on the failing test cases

File: sources/packages/web/src/assets/css/formio-shared.scss (hunk 2 above)
Comment (andrewsignori-aot) [reply sources/packages/web/src/assets/css/formio-shared.scss]: Is it really common across all images? Is it something that does not change depending on where the image is placed?

**Review by ann-aot — APPROVED** [review]
👍 LGTM @henryoforeh-aot

**Review by abschwenker — APPROVED** [review]

**Review by dheepak-aot — APPROVED** [review]
LGTM. Good work @henryoforeh-aot

**Pull request merged**